### PR TITLE
contracts: Add event field names

### PR DESCRIPTION
**Pull request by ascjones**
Use struct variants with field names for `Event`, instead of unnamed tuple variants. This will be automatically reflected (via the `TypeInfo` derive) in the metadata with variant fields having names attached.

Motivated by my having wasted a lot of time when I mixed up these fields `Instantiated(T::AccountId, T::AccountId)`

If this is acceptable then we can convert other pallet `Event`s to this style.

### CODE REVIEW

File: frame/contracts/src/lib.rs [hunk 1]
```diff
@@ -369,49 +369,44 @@ pub mod pallet {
 	#[pallet::event]
 	#[pallet::generate_deposit(pub(super) fn deposit_event)]
 	pub enum Event<T: Config> {
-		/// Contract deployed by address at the specified address. \[deployer, contract\]
```
Comment (ascjones) [inline frame/contracts/src/lib.rs:372]: @jacogr I've removed this `\[deployer, contract\]`, but is that used by `polkadot.js` currently for the field names?

**Review by athei — APPROVED** [review]
LGTM

File: frame/contracts/src/lib.rs (hunk 1 above)
Comment (jacogr) [reply frame/contracts/src/lib.rs:372]: No, it not used at all - it purely appears in the md documentation generated from the metadata, aka https://polkadot.jso.org/docs/.

**Review by shawntabrizi — APPROVED** [review]

File: frame/contracts/src/lib.rs (hunk 1 above)
Comment (shawntabrizi) [reply frame/contracts/src/lib.rs:372]: You will find this annotations sprinkled everywhere. If moving to this field name syntax, I think you can clean them all up :)

**Comment by ascjones** [comment]
bot merge

**Pull request merged**